sklearn/metrics/classification.py

```diff
@@ -1062,7 +1092,12 @@ def _prf_divide(numerator, denominator, metric, modifier, average, warn_for):
         return result
 
     # remove infs
-    result[mask] = 0.0
+    result[mask] = float(zero_division == 1)
```

Comment:
this is obfuscated. I'd rather `0.0 if zero_division in ('warn', 0) else 1`

Reply:
done in new version